docs/reference/migration-guide.mdx

```diff
@@ -0,0 +1,1160 @@
+</Note>
+
+### Multi-Factor Authentication (MFA)
+
+| v3.x | v4.x | Notes |
+|------|------|-------|
+| MFA machine with `generateQrCode()` | `auth.changeUserMfa()` | v3.x used state machines, v4.x uses direct methods |
+| MFA machine with `activateMfa(code)` / `disableMfa(code)` | `auth.verifyChangeUserMfa({ activeMfaType: 'totp', code })` | Activate/disable MFA |
+
+### Social Providers (OAuth)
+
+| v3.x | v4.x | Notes |
+|------|------|-------|
+| `auth.signIn({ provider: 'github' })` | `auth.signInProviderURL('github', { redirectTo })` | Returns URL string |
+| Options in method | Pass `{ connect: accessToken }` | For connecting provider to existing account |
+
+### Storage Methods
+
+| v3.x | v4.x | Notes |
+|------|------|-------|
+| `storage.upload({ file, name })` | `storage.uploadFiles({ "file[]": [file] })` | Different parameter structure |
+| `storage.download({ fileId })` | `storage.getFile(fileId)` | Renamed method, uses positional parameter |
+| `storage.delete({ fileId })` | `storage.deleteFile(fileId)` | Renamed method, uses positional parameter |
+| `storage.getPresignedUrl({ fileId })` | `storage.getFilePresignedURL(fileId)` | Renamed method, uses positional parameter |
+| `storage.getPublicUrl({ fileId })` | `${nhost.storage.baseURL}/files/${fileId}` | Construct URL manually |
+
+### GraphQL Methods
+
+| v3.x | v4.x | Notes |
+|------|------|-------|
+| `graphql.request(query, variables)` | `graphql.request({ query, variables })` | Options object with `query` key |
+
+### Functions Methods
+
+| v3.x | v4.x | Notes |
+|------|------|-------|
+| `functions.call(path, body)` | `functions.post(path, body)` | Renamed method |
+
+---
+
+## `@nhost/react` → `@nhost/nhost-js`
+
+This section covers migrating React applications from the deprecated `@nhost/react` package to using the vanilla JavaScript SDK with custom React patterns.
+
+<Note>
+While this guide uses React, the same changes apply to other frameworks like Next.js and Vue.js. You'll need to adapt the provider patterns and state management to your framework (e.g., using Vue's `provide`/`inject` or composables instead of React Context and hooks).
+</Note>
+
+### Overview of Changes
+
+The old `@nhost/react` package provided:
+- Ready-made hooks (`useSignInEmailPassword`, `useUserData`, etc)
+- Built-in providers (`NhostProvider`, `NhostApolloProvider`)
+- Automatic state management
+
+The new approach requires:
+- Using vanilla `@nhost/nhost-js` SDK
+- Creating your own custom providers and hooks
+- Manual state management with React hooks
+
+### Package Installation
+
+<CodeGroup>
+```bash Old
+npm install @nhost/react @nhost/react-apollo
+```
+
+```bash New
+npm install @nhost/nhost-js
+```
+</CodeGroup>
+
+### Provider Setup
+
+#### Step 1: Create Nhost Provider
+
+Create a context provider to share the Nhost client instance:
+
+```tsx
+// providers/nhost/NhostProvider.tsx
```

Comment:
Why do we have an NhostProvider and an AuthProvider? The NhostProvider doesn't seem to add anything of value. As a matter of fact the NhostProvider and the AuthProvider use completely different nhost client so that feels bit strange